### ISSUE TITLE
[sort-type-constituents] Feature request: support case sensitive sorting

### ISSUE THREAD

**Issue by iliubinskii** [typescript-eslint/typescript-eslint#4602]
**Repro**

```TS
type T = "DeleteForever" | "DeletedAt";
```

**Expected Result**

With new feature should leave as is.

**Actual Result**

There is currently no way to use case sensitive sorting.

**Additional Info**

1. New option will help make this rule consistent with other sorting-related rules.
2. When you have values consisting of several words (like "DeleteForever", "DeletedAt"), case sensitive sorting is more logical. Compare it to kebabcase: "delete-forever", "deleted-at" is more natural.
3. For case-sensitive language (e.g. typescript) case sensitive sorting is good option.

Also, natural number order may also be turned on/off with an option.

**Comment by JoshuaKGoldberg** [comment]
Note that we're feature-freezing `sort-type-constituents` and other naming & sorting stylistic rules: #8792. Since this received a PR (#8760) before #8792 was filed, we'll still accept that PR. But if #8760 is closed then we'll close this issue too.
